src/sonic-config-engine/tests/test_j2files.py

```diff
@@ -336,6 +336,37 @@ def test_buffers_mellanox2410_render_template(self):
         sample_output_file = os.path.join(self.test_dir, 'sample_output', utils.PYvX_DIR, 'buffers-mellanox2410.json')
         assert filecmp.cmp(sample_output_file, self.output_file)
 
+    def test_config_brcm_render_template(self):
+        if utils.PYvX_DIR != 'py3':
+            #Skip on python2 as the change will not be backported to previous version
+            return
+
+        config_bcm_sample_outputs = [
+            'arista7050cx3-dualtor.config.bcm',
+            'arista7260-dualtor.config.bcm',
+            'arista7260-t1.config.bcm'
+        ]
+        sample_minigraph_files = [
+            'sample-arista-7050cx3-dualtor-minigraph.xml',
+            'sample-arista-7260-dualtor-minigraph.xml',
+            'sample-arista-7260-t1-minigraph.xml'
+        ]
+        for i, config in enumerate(config_bcm_sample_outputs):
+            device_template_path = os.path.join(self.test_dir, './data/j2_template')
+            config_sample_output = config_bcm_sample_outputs[i]
+            sample_minigraph_file = os.path.join(self.test_dir,sample_minigraph_files[i])
+            port_config_ini_file = os.path.join(device_template_path, 'port_config.ini')
+            config_bcm_file = os.path.join(device_template_path, 'config.bcm.j2')
+            config_test_output = os.path.join(self.test_dir, 'config_output.bcm')
+
+            argument = '-m ' + sample_minigraph_file + ' -p ' + port_config_ini_file + ' -t ' + config_bcm_file + ' > ' + config_test_output
```

Comment:
I just realized that we are creating our own config_bcm_file template here and using that. Then how do we intend to catch issues when the original .bcm in each of the hwsku folders is modified?

Reply:
we do not. i think the purpose is to mainly to test the template.

i think it is a problem for the unit test to test some files outside of this sonic-config-engine folder, making the sonic-config-engine not self-contained.

Reply:
I think the checks for j2 config, might be divided into two parts, j2 template, and the actual output.

For the j2 file, I chose not to check from the original j2, apart from the self-contained problem, those config files might differ from different platforms, if we want to check them all and make sure the output synced up with the origin file, then we need to copy all those expected output files to the test folder and modify them every time when the origin file changed. Then, in this PR, I only checked the template part.

For checking the content of the generated bcm config, it should be like other .bcm checks. 
It can be checked in the config_checker. Like, illegal/unexpected properties, syntax errors (unexpected token), etc.

Actually, there is an issue with checking the generated bcm config. 
https://github.com/Azure/sonic-buildimage/issues/11042